pkg/avancement/service_manager.go

```diff
@@ -87,6 +90,25 @@ func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName stri
 		}
 	}(keepCache, &reposToDelete)
 
+	var localSource git.Source
+	var errorSource error
+	if fromLocalRepo(fromURL) {
+		localSource = s.localFactory(fromURL, s.debug)
+	} else {
+		source, errorSource = s.checkoutSourceRepo(fromURL, fromBranch)
+		if errorSource != nil {
+			return fmt.Errorf("failed to checkout repo: %w", errorSource)
+		}
+		reposToDelete = append(reposToDelete, source)
+	}
+
+	if newBranchName == "" {
```

Comment:
How about extracting a generateBranchName() method, and adding a unit test for it?

Reply:
that seems like a good idea, I'll do that now :)